tokens/test/erc-1155/ERC1155Rarible.test.js

```diff
@@ -30,6 +30,485 @@ contract("ERC1155Rarible", accounts => {
+      let minter = accounts[1];
+      let anotherUser = accounts[6];
+      let transferTo = accounts[2];
+
+      const tokenId4 = minter + "b00000000000000000000004"; //save token creator
+      const tokenId5 = transferTo + "b00000000000000000000005"; //save token creator
+      const tokenURI = "/uri";
+      let supply = 5;
+      let mintValue = 5;
+      await expectThrow(  //throw transferTo not owner for tokenId4
+        token.burnBatch(transferTo, [tokenId4, tokenId5], [2, 2], {from: transferTo})
+      );
+      await token.burnBatch(transferTo, [tokenId5], [2], {from: transferTo});
+      await token.burnBatch(minter, [tokenId4], [2], {from: minter});
+      await expectThrow( //can`t mint 5, 2 already burned
+        token.mintAndTransfer([tokenId4, tokenURI, supply, creators([minter]), [], [zeroWord]], transferTo, 5, {from: minter})
+      );
+      await token.mintAndTransfer([tokenId5, tokenURI, supply, creators([transferTo]), [], [zeroWord]], transferTo, 3, {from: transferTo});
+      assert.equal(await token.balanceOf(transferTo, tokenId5), 3);
+    });
+
+    it("BurnBatch two tokens, ok", async () => {
+      let minter = accounts[1];
+      let anotherUser = accounts[6];
+      let transferTo = accounts[2];
+
+      const tokenId4 = minter + "b00000000000000000000004"; //save token creator
+      const tokenId5 = transferTo + "b00000000000000000000005"; //save token creator
+      const tokenURI = "/uri";
+      let supply = 5;
+      let mintValue = 5;
+      //mint, after do burnBatch
+      await token.mintAndTransfer([tokenId4, tokenURI, supply, creators([minter]), [], [zeroWord]], transferTo, mintValue, {from: minter})
+      assert.equal(await token.balanceOf(transferTo, tokenId4), mintValue);
+      //combine minted and not yet minted, try to burnBatch
+      await token.burnBatch(transferTo, [tokenId4, tokenId5], [mintValue, mintValue], {from: transferTo})
+      assert.equal(await token.balanceOf(transferTo, tokenId4), 0);//burn all minted tokenId4, burn all lazy tokenId5
+      await expectThrow(  //no lazy tokenId5, throw
+        token.mintAndTransfer([tokenId5, tokenURI, supply, creators([minter]), [], [zeroWord]], anotherUser, 1, {from: transferTo})
+      );
+    });
+
+    it("BurnBatch three tokens, ok", async () => {
+      let minter = accounts[1];
+      let anotherUser = accounts[6];
+      let transferTo = accounts[2];
+
+      const tokenId1 = minter + "b00000000000000000000001"; //save token creator
+      const tokenId2 = minter + "b00000000000000000000002"; //save token creator
+      const tokenId3 = minter + "b00000000000000000000003"; //save token creator
+      const tokenURI = "/uri";
+      let supply = 5;
+      let burn = 2;
+      let mintValue = 5;
+      //BurnBatch and after do mint
+      await expectThrow(
+        token.burnBatch(anotherUser, [tokenId1, tokenId2, tokenId3], [burn, burn, burn], {from: anotherUser})  //token has another creator
```

Comment:
we should probably try to send this transaction from minter to check the specific require

Reply:
ok, add this check